x-pack/plugins/ml/public/embeddables/anomaly_swimlane/explorer_swimlane_container.tsx

```diff
@@ -0,0 +1,289 @@
+import { CoreStart } from 'kibana/public';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { ExplorerSwimlane } from '../../application/explorer/explorer_swimlane';
+import { TimeBuckets } from '../../application/util/time_buckets';
+import { ExplorerJob, OverallSwimlaneData } from '../../application/explorer/explorer_utils';
+import { MlStartDependencies } from '../../plugin';
+import {
+  AnomalySwimlaneEmbeddableInput,
+  AnomalySwimlaneEmbeddableOutput,
+  MlServices,
+} from './anomaly_swimlane_embeddable';
+import { parseInterval } from '../../../common/util/parse_interval';
+import { SWIMLANE_TYPE } from '../../application/explorer/explorer_constants';
+import { MlTooltipComponent } from '../../application/components/chart_tooltip';
+import { Filter } from '../../../../../../src/plugins/data/common/es_query/filters';
+import { esKuery, Query } from '../../../../../../src/plugins/data/public';
+
+const RESIZE_THROTTLE_TIME_MS = 500;
+
+export interface ExplorerSwimlaneContainerProps {
+  id: string;
+  embeddableInput: Observable<AnomalySwimlaneEmbeddableInput>;
+  services: [CoreStart, MlStartDependencies, MlServices];
+  refresh: Observable<any>;
+  onOutputChange: (output: Partial<AnomalySwimlaneEmbeddableOutput>) => void;
+}
+
+export const ExplorerSwimlaneContainer: FC<ExplorerSwimlaneContainerProps> = ({
+  id,
+  embeddableInput,
+  services,
+  refresh,
+}) => {
+  const [{ uiSettings }, , { explorerService, mlAnomalyDetectorService }] = services;
+
+  const [swimlaneData, setSwimlaneData] = useState<OverallSwimlaneData>();
+  const [chartWidth, setChartWidth] = useState<number>(0);
+  const [swimlaneType, setSwimlaneType] = useState<string>();
+  const [error, setError] = useState();
+
+  const chartWidth$ = useMemo(() => new Subject<number>(), []);
+
+  const timeBuckets = useMemo(() => {
+    return new TimeBuckets({
+      'histogram:maxBars': uiSettings.get('histogram:maxBars'),
+      'histogram:barTarget': uiSettings.get('histogram:barTarget'),
+      dateFormat: uiSettings.get('dateFormat'),
+      'dateFormat:scaled': uiSettings.get('dateFormat:scaled'),
+    });
+  }, []);
+
+  const jobs$ = useMemo(
+    () =>
+      embeddableInput.pipe(
+        pluck('jobIds'),
+        distinctUntilChanged(isEqual),
+        switchMap(jobsIds => mlAnomalyDetectorService.getJobs$(jobsIds))
+      ),
+    []
+  );
+
+  useEffect(() => {
+    chartWidth$.next(chartWidth);
+  }, [chartWidth]);
+
+  useEffect(() => {
+    const subscription = combineLatest([
+      jobs$,
+      embeddableInput,
+      chartWidth$.pipe(
+        skipWhile(v => !v),
+        distinctUntilChanged((prev, curr) => {
+          // emit only if the width has been changed significantly
+          const allowedDiff = 20;
+          const diff = Math.abs(curr - prev);
+          return diff < allowedDiff;
+        })
+      ),
+      refresh.pipe(startWith(null)),
+    ])
+      .pipe(
+        debounceTime(500),
```

Comment:
Can this use `RESIZE_THROTTLE_TIME_MS ` instead of the hardcoded `500`?

Reply:
throttle and debounce serve a different purpose here, but I'll create a constant instead of using a magic number

Reply:
https://github.com/elastic/kibana/pull/64056/commits/5801a72accf0b20097aefba46ac6183b2a5bd1c4